spec/graphql/types/query_type_spec.rb

```diff
@@ -1,32 +1,277 @@
+          ids = response_data[:mediaEntries][:edges].map { |n| n[:node][:id] }
+
+          expect(response_data[:mediaEntries][:edges].length).to eq(first)
+          expect(ids).to eq(collection.media_entries.offset(2).take(2).pluck(:id))
+        end
+
+        it 'contains MediaEntries ordered as speficied in query' do
+          fill_collection_with_media_entries(collection)
+
+          query = collection_query(
+            collection.id,
+            first: collection.media_entries.length,
+            order_by: 'CREATED_AT_ASC')
+          response_data = response_as_hash(query)[:data][:collection]
+
+          ids = response_data[:mediaEntries][:edges].map { |n| n[:node][:id] }
+          ordered_media_entries = collection.media_entries.order('created_at ASC')
+
+          expect(ids).to eq(ordered_media_entries.ids)
+        end
+
+        it 'contains pageInfo for mediaEntries collection' do
+          expect(response_data[:mediaEntries][:pageInfo].keys).
+            to eq(%w(endCursor startCursor hasPreviousPage hasNextPage))
+        end
+      end
+    end
+
+    def fill_collection_with_media_entries(collection)
+      collection.media_entries = FactoryGirl.create_list(:media_entry_with_title,
+                                                          4)
     end
 
-    it 'defines a field title of String type' do
-      expect(subject).to have_field(:id).that_returns(type('String'))
+    def stringified_created_ats_from_response(response)
+      response.map do |media_entry|
+        ActiveSupport::TimeZone['UTC'].parse(media_entry[:createdAt]).to_s
+      end
     end
 
+   # todo 
+   # move queries to helper
+
+    def media_entry_query(id)
+      <<-GRAPHQL
+        {
+          mediaEntry(id: "#{id}") {
+            id
+            createdAt
+            title
+          }
+        }
+      GRAPHQL
+    end
 
+    def collection_query(id, first: 2, cursor: nil, order_by: nil)
+      <<-GRAPHQL
+        {
+          collection(id: "#{id}") {
+            id
+            getMetadataAndPreviews
+            createdAt
+            updatedAt
+            layout
+            sorting
+            responsibleUserId
+            mediaEntries(first: #{first}
+                         #{', after: ' if cursor} #{cursor}
+                         #{', orderBy: ' if order_by} #{order_by}) {
+              pageInfo {
+                endCursor
+                startCursor
+                hasPreviousPage
+                hasNextPage
+              }
+              edges {
+                node {
+                  id
+                  createdAt
+                  title
+                }
+              }
+            }
+          }
+        }
+      GRAPHQL
+
+    end
+
+    def media_entries_query(first: nil, order_by: nil)
+      first = "first: #{first}" if first
+      order_by = "orderBy: #{order_by}" if order_by
+      params = [first, order_by].join(', ')
+
+      <<-GRAPHQL
+        {
+          allMediaEntries(#{params}) {
```

Comment:
@kooropatfa instead of string interpolations I'd prefer to use `query variables` designed for this purpose , please refer to https://graphql-ruby.org/queries/executing_queries.html#variables :)

Reply:
i agree, could not have said it better :)